viewing_party/party.py

```diff
@@ -1,23 +1,219 @@
+import statistics
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
-    pass
+def create_movie(title=None, genre=None, rating=None):
+    """
+    Creates a dictionary with movie details
+    :params: str: title - movie title; default=None
+             str: genre - movie genre; default=None
+             str: rating - movie rating; default=None
+    :returns: new_movie - dictionary
+    """
+    dets = [title, genre, rating]
+    # create dictionary with movie details
+    new_movie = {}
+    keys = ["title", "genre", "rating"]
+    for key, det in zip(keys, dets):
+        new_movie[key] = det
+    # handle missing movie details
+    if any(x == None for x in dets):
+        new_movie = None
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    """
+    Updates dataset to include watched movies and details
+    :params: dict - user_data: "watched" movies
+             dict - movie: movie details
+    :returns: nested dict - updated_data: updated watched movies 
+    """
+    # copy user_data
+    updated_data = user_data.copy()
+    # update watched movie
+    updated_data["watched"] = [movie]
+    return updated_data
+
+
+def add_to_watchlist(user_data, movie):
+    """
+    Updates dataset to include movies to be watched and details
+    :params: dict - user_data: "watchlist" movies
+             dict - movie: movie details
+    :returns: nested dict - updated_data: updated watchlist movies
+    """
+    # copy user data
+    updated_data = user_data.copy()
+    # update watchlist movie
+    updated_data["watchlist"] = [movie]
+    return updated_data
+
+def watch_movie(janes_data, MOVIE_TITLE_1):
+    """
+    Moves a movie from watchlist to watched
+    :params: dict - janes_data: dataset containing watched and watchlist movies
+             str - MOVIE_TITLE_1: movie title
+    :returns: nested dict - updated_data: updated watched/watchlist movies
+    """
+    # copy user data
+    updated_data = janes_data.copy()
+    # iterate through watchlist movies
+    for movie in updated_data['watchlist']:

```

Comment:
This code passes the tests, but as an aside it contains an "anti-pattern" that is important to be aware of.  An element of the list that is being iterated over is being removed inside the for loop.  This is generally considered bad practice and can leave you open to problems in more complex code.  One solution could be to copy the data you get from "movie in updated_data['watchlist']" into a variable that's not going to change as you modify updated_data.